src/main/webapp/app/pages/curation/list/FirebaseList.tsx

```diff
@@ -3,19 +3,36 @@ import { componentInject } from 'app/shared/util/typed-inject';
 import { IRootStore } from 'app/stores';
 import { observer } from 'mobx-react';
 import { onValue, ref } from 'firebase/database';
-import { ViewportList, ViewportListPropsBase } from 'react-viewport-list';
 
-export interface IFirebaseListProps<T> extends StoreProps, ViewportListPropsBase {
+export interface IFirebaseListProps<T> extends StoreProps {
   path: string;
   itemBuilder: (firebaseIndex: number) => React.ReactNode;
   pushDirection: 'front' | 'back';
   filter?: (firebaseIndex: number) => boolean;
   defaultSort?: (a: T, b: T) => number;
+  onInitialRender?: () => void;
 }
 
-function FirebaseList<T>({ path, itemBuilder, pushDirection, filter, defaultSort, firebaseDb, ...rest }: IFirebaseListProps<T>) {
+function FirebaseList<T>({ path, itemBuilder, pushDirection, filter, defaultSort, onInitialRender, firebaseDb }: IFirebaseListProps<T>) {
   const [indices, setIndices] = useState<number[]>(null);
   const [numItemsAdded, setNumItemsAdded] = useState(0);
+  const [initialRenderComplete, setInitialRenderComplete] = useState(false);
+
+  useEffect(() => {
+    // https://webperf.tips/tip/measuring-paint-time/
+    if (!onInitialRender || initialRenderComplete || !indices) {
+      return;
+    }
+
+    requestAnimationFrame(() => {
+      const messageChannel = new MessageChannel();
+      messageChannel.port1.onmessage = () => {
+        onInitialRender();
+        setInitialRenderComplete(true);
+      };
+      messageChannel.port2.postMessage(undefined);
+    });
+  }, [indices, initialRenderComplete]);
```

Comment:
Is waiting for the UI to be painted only necessary because we are now loading all mutations and that takes some time? Is it negligible when we render lets say 10 or 20 mutations only (without viewport)?

Reply:
Waiting for the UI to be painted was never negligible. The time between the component mounting and the list being drawn has always taken some non-negligible amount of time. After mounting, regardless of our approach, we need to fetch and process all mutations. The time to draw all mutations (without viewport) is slightly longer than only 10 or 20 mutations, but not significantly.

I think measuring the draw time is a good approach, because it accounts for the time between mounting and firebase updates (and drawing) on the slowest component, so the user will never see the data blink.

Reply:
Just measured with the library and we are seeing paint times of 1ms to 2ms.